layers/+lang/ruby/funcs.el

```diff
@@ -10,6 +10,56 @@
 ;;; License: GPLv3
 
 
+;; backend
+(defun spacemacs//ruby-setup-backend ()
+  "Conditionally configure Ruby backend"
+  (pcase ruby-backend
+    (`lsp (spacemacs//ruby-setup-lsp))
+    (`robe (spacemacs//ruby-setup-robe))))
+
+(defun spacemacs//ruby-setup-company ()
+  "Configure backend company"
+  (pcase ruby-backend
+    (`lsp (spacemacs//ruby-setup-lsp-company))
+    (`robe (spacemacs//ruby-setup-robe-company))))
+
+
+;; lsp
+
+(defun spacemacs//ruby-setup-lsp ()
+  "Setup Ruby lsp."
+  (if (configuration-layer/layer-used-p 'lsp)
+      (lsp)
+    (message "`lsp' layer is not installed, please add `lsp' layer to your dotfile.")))
+
+(defun spacemacs//ruby-setup-lsp-company ()
```

Comment:
no longer needed, lsp-mode handles that automatically.

Reply:
To clarify, is this entire ruby-setup-lsp-company function not needed?

Reply:
Yes. lsp-mode automatically configures company-lsp if it is present.